src/networking.c

```diff
@@ -2546,7 +2577,7 @@ void readQueryFromClient(connection *conn) {
     if (c->reqtype == PROTO_REQ_MULTIBULK && c->multibulklen && c->bulklen != -1
         && c->bulklen >= PROTO_MBULK_BIG_ARG)
```

Comment:
i think you should add a `CLIENT_MASTER` check here too.
the implications if we don't:
1. we'll still attempt to avoid reading any bytes belonging to the next arg (which is not a concern now, so we can relax it).
2. we'll flag the `big_arg` variable, and that'll lead to a non-greedy resizing of the query buffer (which is also not a concern now)

Reply:
commit [a5df94a](https://github.com/redis/redis/pull/10413/commits/a5df94a2e601446dc1375b976fc09d78fd624e87)